backend/libinput/pointer.c

```diff
@@ -141,6 +141,54 @@ void handle_pointer_axis(struct libinput_event *event,
 	wlr_signal_emit_safe(&wlr_dev->pointer->events.frame, wlr_dev->pointer);
 }
 
+void handle_pointer_axis_wheel(struct libinput_event *event,
+		struct libinput_device *libinput_dev) {
+	struct wlr_input_device *wlr_dev =
+		get_appropriate_device(WLR_INPUT_DEVICE_POINTER, libinput_dev);
+	if (!wlr_dev) {
+		wlr_log(WLR_DEBUG, "Got a pointer event for a device with no pointers?");
+		return;
+	}
+	struct libinput_event_pointer *pevent =
+		libinput_event_get_pointer_event(event);
+	struct wlr_event_pointer_axis wlr_event = { 0 };
+	wlr_event.device = wlr_dev;
+	wlr_event.time_msec =
+		usec_to_msec(libinput_event_pointer_get_time_usec(pevent));
+
+	if (libinput_event_pointer_get_axis_source(pevent) != LIBINPUT_POINTER_AXIS_SOURCE_WHEEL) {
+		wlr_log(WLR_DEBUG, "Got a wheel axis event for a non-wheel device");
+		return;
+	}
+	wlr_event.source = WLR_AXIS_SOURCE_WHEEL;
+
+	const enum libinput_pointer_axis axes[] = {
+		LIBINPUT_POINTER_AXIS_SCROLL_VERTICAL,
+		LIBINPUT_POINTER_AXIS_SCROLL_HORIZONTAL,
+	};
+	for (size_t i = 0; i < sizeof(axes) / sizeof(axes[0]); ++i) {
+		if (libinput_event_pointer_has_axis(pevent, axes[i])) {
+			switch (axes[i]) {
+			case LIBINPUT_POINTER_AXIS_SCROLL_VERTICAL:
+				wlr_event.orientation = WLR_AXIS_ORIENTATION_VERTICAL;
+				break;
+			case LIBINPUT_POINTER_AXIS_SCROLL_HORIZONTAL:
+				wlr_event.orientation = WLR_AXIS_ORIENTATION_HORIZONTAL;
+				break;
+			}
+			double hr_delta =
+					libinput_event_pointer_get_axis_value_v120(pevent, axes[i]);
+			if (hr_delta != 0) {
```

Comment:
`hr_delta` should never be zero, you wouldn't get a `AXIS_WHEEL` event otherwise.

Reply:
Should or must never? The check is just because it would lead to a divide-by-zero error if it ever was.

Reply:
for libinput at least, I don't see a legitimate case of sending those events with a delta of zero (one axis maybe, but not the other one). but feel free to leave it there, better safe than sorry I guess.

Reply:
If you're happy I'll never see a zero there I'm be happy to remove it, one less line run on every input is fine with me!

Reply:
Could just `assert(delta != 0)` instead

Reply:
Good thinking, thanks.